script/commands/register.ts

```diff
@@ -0,0 +1,149 @@
+registerCmd.command('ism-multisig').action(handleRegisterIsm);
+
+type AggregateHook = typed<'hpl_hook_aggregate'> & {
+  hooks: ContextHook[];
+};
+
+type IgpHook = typed<'hpl_igp'> & { oracle: typed<'hpl_igp_oracle'> };
+
+function isIgpHookType(hook: ContextHook): hook is IgpHook {
+  return hook.type === 'hpl_igp';
+}
+
+function findIsmsConfig() {
+  return config.deploy.ism;
+}
+
+function findIgpHookInAggregate() {
+  const defaultHook = config.deploy.hooks?.default;
+  if (defaultHook && defaultHook.type === 'aggregate') {
+    const igpHook = defaultHook.hooks.find(
+      (hook): hook is IgpHookType => hook.type === 'igp'
+    );
+    return igpHook;
+  }
+  return undefined;
+}
+
+async function handleRegisterIsm(_: object, cmd: Command) {
+  const { ctx, client } = CONTAINER.get(Dependencies);
+
+  const ismMultisig = ctx.deployments.isms;
+  let ismMultisigAddress: string | undefined;
+  if (ismMultisig && ismMultisig.type === 'hpl_ism_multisig') {
+    ismMultisigAddress = ismMultisig.address;
+  }
+  if (!ismMultisigAddress) {
+    throw new Error('Ism multisig context not found');
+  }
+  const ismsConfig = findIsmsConfig();
+  if (!ismsConfig) {
+    throw new Error('Ism multisig config not found');
+  }
+  switch (ismsConfig.type) {
+    case 'multisig': {
+      const multisig = {
+        type: 'hpl_ism_multisig',
+        address: ismMultisigAddress as string,
+      };
+      await executeMultiMsg(
+        client,
+        Object.entries(ismsConfig.validators).map(([domain, { addrs, threshold }]) => ({
+          contract: multisig,
+          msg: {
+            set_validators: {
+              domain: Number(domain),
+              threshold,
+              validators: addrs.map((v) =>
+                v === '<signer>' ? client.signer_addr : v
+              ),
+            },
+          },
+        }))
+      );
+      break;
+    }
+    default:
+      throw new Error('Multisig ism config not found');
+  }
+}
+
+async function handleRegisterIgpOracle(_: object, cmd: Command) {
+  const { ctx, client } = CONTAINER.get(Dependencies);
+
+  const defaultHook = ctx.deployments.hooks?.default;
+  let igpHook;
+  let aggregateHook;
+  let igpOracleAddress: string | undefined;
+  if (defaultHook && defaultHook.type === 'hpl_hook_aggregate') {
+    igpHook = defaultHook.hooks.find(isIgpHookType) as IgpHook;
+    igpOracleAddress = igpHook?.oracle?.address;
+    aggregateHook = defaultHook as AggregateHook;
+  }
+  if (!igpHook) {
+    throw new Error('igpHook is undefined in context');
+  }
+  if (!igpOracleAddress) {
+    throw new Error('igpOracleAddress is undefined in context');
+  }
+  if (!aggregateHook) {
+    throw new Error('aggregateHook is undefined in context');
+  }
+
+  const igpType: IgpHook | undefined = aggregateHook?.hooks.find(
+    (hook): hook is IgpHook => hook.type === 'hpl_igp'
+  );
+  if (!igpType) {
+    throw new Error('igpType is undefined');
+  }
+
+  const igpHookConfig = findIgpHookInAggregate();
```

Comment:
I found myself conflating the config vs context when reviewing this. I like that you defined this one with `Config` - I think it'd be good to name the above ones similarly (e.g. `igpHookDeployment`)

One thing you could do is just bundle this validation in the helper`findIgpHookInAggregate` so that it does the above validation and then returns the address and config

Reply:
it's kind of painful, yea